BenmoussaYounes/chat-app
# 9- feature integrate chat api ✨

## JiraTicket [FI-9](https://contributors.atlassian.net/jira/software/projects/ATX/boards/6?selectedIssue=ATX-9)

## Description 📑
- integrate chat api functionality using firestore service 

## Type of Change


## Changes

- integrate chat api, using firestore service
- update conversation model to show the 2 user different notification count
- setup dependency injection using getIt
- add new user enum

This pull request includes the following changes:

## Screenshots 📷

| 01                                    | 02                                    |
| ------------------------------------- | ------------------------------------- | 
| <img src="https://github.com/user-attachments/assets/5d4df553-89df-4212-a4f3-1c02c936fcb0" width="100" height="250"> | <img src="https://github.com/user-attachments/assets/bbb2e8dd-08df-46f8-b89e-24721d74d765" width="100" height="250"> |

## How to test 🚦
- run the app, use the + floating action button to create a new conversation wich should be accesible for both pages
- update unseen messages count from firebase, should result in real time updates for the notification icon on the app dynamically for both users
